Assignment 3-JS/script/script.js

```diff
@@ -0,0 +1,320 @@
+
+Date.prototype.getWeek = function (dowOffset) {
+  /*getWeek() was developed by Nick Baicoianu at MeanFreePath: http://www.meanfreepath.com */
+      dowOffset = typeof(dowOffset) == 'number' ? dowOffset : 0; //default dowOffset to zero
+      var newYear = new Date(this.getFullYear(),0,1);
+      var day = newYear.getDay() - dowOffset; //the day of week the year begins on
+      day = (day >= 0 ? day : day + 7);
+      var daynum = Math.floor((this.getTime() - newYear.getTime() - 
+      (this.getTimezoneOffset()-newYear.getTimezoneOffset())*60000)/86400000) + 1;
+      var weeknum;
+      //if the year starts before the middle of a week
+      if(day < 4) {
+          weeknum = Math.floor((daynum+day-1)/7) + 1;
+          if(weeknum > 52) {
+              nYear = new Date(this.getFullYear() + 1,0,1);
+              nday = nYear.getDay() - dowOffset;
+              nday = nday >= 0 ? nday : nday + 7;
+              /*if the next year starts before the middle of
+                the week, it is week #1 of that year*/
+              weeknum = nday < 4 ? 1 : 53;
+          }
+      }
+      else {
+          weeknum = Math.floor((daynum+day-1)/7);
+      }
+      return weeknum;
+  };
+  var title = document.getElementById('modal-title');
+  var body = document.querySelector('.modal-body p');
+  var event_date = document.getElementById('event-date');
+  function triggerModal(obj){
+    console.log(obj);
+    title.innerHTML = obj.title;
+    body.innerHTML = obj.discription;
+    event_date.innerHTML = obj.date;
+    $('#myModal').modal('show');//check callback /update data
+    $('#myModal').modal('hide');
+  }
+
+//to open side-panel
+document.getElementsByClassName('current-year-name')[0].addEventListener('click', () => {
+  document.getElementById("mySidepanel").style.height = "20%";
+});
+//to hide the side-panel
+ function closeNav() {
+    document.getElementById("mySidepanel").style.height = "0";
+}
+
+//to render years
+  function Years(){
+    let d = new Date();
+    let currentyear = d.getFullYear();
+    
+      for(let i = 2050;i >= 1950; i--) {
+        const div = document.createElement("div");
+        div.classList.add('year-name');
+        const node = document.createTextNode(i);
+        div.appendChild(node);
+        const element = document.getElementById("years");
+        element.appendChild(div);
+        if(i == currentyear){
+          div.classList.add('current-year');
+          let yearElement = document.querySelector('.current-year');
+          yearElement.scrollIntoView();
+          yearElement.classList.add('firstElement');
+          yearElement.style.color = " #00e6b8 ";
+        }
+      }
+  }
+
+let arrow_up = document.getElementById('icon1');
+
+arrow_up.addEventListener('click',function(){
+  var Element = document.querySelector('.firstElement');
```

Comment:
Revisit all the variable's name

Reply:
Changed variable names to more readable names.